django_saml2_auth/user.py

```diff
@@ -192,31 +195,69 @@ def create_jwt_token(user_id: str) -> Optional[str]:
         user_id (str): User's username or email based on User.USERNAME_FIELD
 
     Raises:
-        SAMLAuthError: Cannot create JWT token. Specify secret and algorithm.
+        SAMLAuthError: Cannot create JWT token. Specify an algorithm.
+        SAMLAuthError: Cannot create JWT token. Specify a secret.
+        SAMLAuthError: Cannot create JWT token. Specify a private key.
 
     Returns:
         Optional[str]: JWT token
     """
     user_model = get_user_model()
 
-    jwt_secret = settings.SAML2_AUTH.get("JWT_SECRET")
     jwt_algorithm = settings.SAML2_AUTH.get("JWT_ALGORITHM")
+    jwt_secret = settings.SAML2_AUTH.get("JWT_SECRET")
+    jwt_private_key = settings.SAML2_AUTH.get("JWT_PRIVATE_KEY")
+    jwt_private_key_passphrase = settings.SAML2_AUTH.get("JWT_PRIVATE_KEY_PASSPHRASE")
     jwt_expiration = settings.SAML2_AUTH.get("JWT_EXP", 60)  # default: 1 minute
     payload = {
         user_model.USERNAME_FIELD: user_id,
         "exp": (datetime.utcnow() +
                 timedelta(seconds=jwt_expiration)).timestamp()
     }
 
-    if not jwt_secret or not jwt_algorithm:
-        raise SAMLAuthError("Cannot create JWT token. Specify secret and algorithm.", extra={
+    if not jwt_algorithm:
+        raise SAMLAuthError("Cannot create JWT token. Specify an algorithm.", extra={
+            "exc_type": Exception,
+            "error_code": NO_JWT_ALGORITHM,
+            "reason": "Cannot create JWT token for login.",
+            "status_code": 500
+        })
+
+    if (jwt_algorithm in ["HS256", "HS384", "HS512"] and
+            not jwt_secret):
+        raise SAMLAuthError("Cannot create JWT token. Specify a secret.", extra={
             "exc_type": Exception,
-            "error_code": NO_JWT_SECRET_OR_ALGORITHM,
+            "error_code": NO_JWT_SECRET,
+            "reason": "Cannot create JWT token for login.",
+            "status_code": 500
+        })
+    elif (jwt_algorithm not in ["HS256", "HS384", "HS512"] and
```

Comment:
Super minor nitpick, could this be `if` instead of `elif`? It makes the reading a bit easier as you know this has nothing to do with previous logic block.